### PR TITLE
[OTEL-1673] Improve host metadata resource attributes

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** #5248 but now hopefully without crashes. Reverts #5249

### CODE REVIEW

**Comment by mx-psi** [comment]
@mackjmr The `dd-gitlab/staging-deploy-ds-gateway-eks` job failed, but I am not sure if it is related to this PR

```
Error: UPGRADE FAILED: rendered manifests contain a resource that already exists. Unable to continue with update: ClusterRole "opentelemetry-collector-ds" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; annotation validation error: key "meta.helm.sh/release-name" must equal "opentelemetry-collector": current value is "opentelemetry-collector-ds"
```

Is this related to my changes?

File: ci/values.yaml [hunk 1]
```diff
@@ -97,6 +103,15 @@ config:
       - key: service
         from_attribute: service_name
         action: upsert
+    transform:
```
Comment (mackjmr) [inline ci/values.yaml:106]: This is fine, but why not use a resource processor instead ? Seems like we are just trying to insert resource attributes, and don't have some complex transformation that requires transform processor ?

**Review by mackjmr — APPROVED** [review]

File: ci/values.yaml (hunk 1 above)
Comment (mx-psi) [reply ci/values.yaml:106]: Discussed offline, given the possibility that we will drop the resource processor in the future this seems more future proof to me.

**Comment by mx-psi** [comment]
Will merge this on Monday morning :)

**Pull request merged**